src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -863,19 +864,17 @@ internal async Task HandleCopyResultsRequest(CopyResultsRequestParams requestPar
                             }
                         }
                         // Add line break if this is not the last row in all selections.
-                        if (rowIndex + pageStartRowIndex != lastRowIndex && (!builder.ToString().EndsWith(Environment.NewLine) || (!Settings?.QueryEditorSettings?.Results?.SkipNewLineAfterTrailingLineBreak ?? true)))
+                        var lastChar = builder[builder.Length - 1];
```

Comment:
This implementation will behave differently from the old implementation on windows when the string ends with \n but not \r\n. (Old implementation would return false for the `EndsWith` check but new one would return true).

Instead of doing manual char level comparison, you can probably use [ToString(int,int)](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.tostring?view=net-8.0&redirectedfrom=MSDN#system-text-stringbuilder-tostring(system-int32-system-int32)) overload, which should still be able to avoid the perf overhead of full string generation of `ToString()`.

```
public static bool EndsWith(StringBuilder sb, string test)
{
    if (sb.Length < test.Length)
        return false;
    return sb.ToString(sb.Length - test.Length, test.Length).EndsWith(test);
}
```